New buchhalter commands: `vault list`, `vault remove`, `vault select` and `vault add`

andygrunwald opened the pull request:
## Context

This Pull Request adds 4 new commands to manage the buchhalter-cli configuration about its 1Password vaults:

* `buchhalter vault list`: Listing all configured vaults
* `buchhalter vault remove`: Remove one local configured vault from buchhalter-cli configuration
* `buchhalter vault select`: Select one local configured vault from buchhalter-cli configuration as new default
* `buchhalter vault add`: Add or override a 1Password vault in the local buchhalter-cli configuration

This Pull Request removes 1 existing command from buchhalter-cli:

* `buchhalter disconnect`: This command is not needed anymore as we are now able to manage multiple vaults in the configuration (and not only one as this command was doing)

This Pull Request adds additional features:

* Add a `--vault=<name>` parameter into the `buchhalter sync` command
* Add a check if we have a default vault selected into the `buchhalter sync` command

## TODOs

This Pull Request is not complete yet. Here is a list of what is missing:

1. Add support to add the Buchhalter SaaS API Key during the `add` operation
2. Remove the `buchhalter connect` command
5. Update local documentation about the new commands

brgmn commented:
@andygrunwald Just checked what you built until now and everything seems to work. The only thing I would change are these many red X icons for not connected vaults. I would change it to something friendlier - e.g. a simple gray icon.

andygrunwald commented:
@brgmn I replaced the  error emojy with empty checkbox emojy in https://github.com/buchhalter-ai/buchhalter-ai-cli/pull/122/commits/22499e4f6f537d5b862b71e39677f64be6be5c46

Additionally, I added a new flag `--vault` to `buchhalter sync` cmd in https://github.com/buchhalter-ai/buchhalter-ai-cli/pull/122/commits/ca5c6b54940e307fbc3437f973f03e0d60b2aff4
